### DISCUSSION TITLE
Systematic building failure on the `binutils` branch

### DISCUSSION THREAD

**Discussion by T0T4R4** [wezm/Retro68#1] [General]
Hi Wez, 

thanks for enabling the discussion tab 👍🏻 

I have attempted during the last 8 hours to build the `binutils` branch, but in the best scenario, the following error breaks the toolchain build (note that I managed to build the `master` branch , but there is something really wrong with the `binutils` one, and I have no idea what it is):
```
/root/project/gcc/libgcc/crtstuff.c:426:1: internal compiler error: in fixup_args_size_notes, at expr.c:4130
```

Would you mind sharing with us the environment required to build this branch (as this is a requirement for the [Classic Mac Rust](https://github.com/T0T4R4/classic-mac-rust) repo ?

Linux distro ?
GCC / G++ version ?
anything else ;) ?

Any **actually** working DockerFile or similar...

Thanks heaps !!
Mathieu

**Comment by wezm** [comment]
My `binutils` branch is ahead of upstream `master` by only one commit, which I brought in from the upstream `gcc12-update` branch. I built it on Arch Linux probably with gcc 12. The version I currently have installed is: `gcc (GCC) 12.2.1 20230201`. However, it may have been updated since I did the Retro68 build. In my personal notes I have the following, which may be useful (or not). Of note I'm only building the PPC toolchain.

- Check out submodules
- Install universal headers
- create build dir and cd into it
- `../Retro68/build-toolchain.bash --universal --no-68k --no-carbon`
